File: Omin/functions/1.py
def addInterest(balance, rate):
    for i in range(len(balance)):
        balance[i] = balance[i] * (1 + rate)
        
def test():
    amounts = [1000, 2200, 800, 360]
    rate = 0.05
    addInterest(amounts, rate)
    print(amounts)
    
test() # [1050.0, 2310.0, 840.0, 378.0]

File: functions/1.cpp
#include <iostream>


void addInterest(float balance, float rate) {
    balance += balance * rate;
    std::cout << "New balance: " << balance << std::endl;
    std::cout << "Address of balance: " << &balance << std::endl;
}

int main() {
    float amount = 1000;
    float rate = 0.05;
    addInterest(amount, rate);
    std::cout<< "Balance after interest: " << amount << std::endl;
    return 0;
}